src/bin/cargo/commands/remove.rs

```diff
@@ -109,9 +110,14 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
 
         // Reload the workspace since we've changed dependencies
         let ws = args.workspace(config)?;
-        resolve_ws(&ws)?;
-    }
+        let (_, resolve) = resolve_ws(&ws)?;
 
+        // Attempt to gc unused patches and re-resolve if anything is removed
+        if gc_unused_patches(&workspace, &resolve)? {
+            let ws = args.workspace(config)?;
+            resolve_ws(&ws)?;
```

Comment:
I've not looked at the lockfile for one of these but I take it the lockfile needs to be updated when removing an unused patch?

Should we include the lockfile in the test so we can make sure this continues working?

Reply:
Hmm this particular test case doesn't actually have unused patches populated. I can make the test so that it does though. Is it better for that to be a separate test?

Reply:
We have an existing test for patch gc, so it could be in there